plugins/vpc-shared-eni/network/bridge_linux.go

```diff
@@ -229,7 +229,26 @@ func (nb *BridgeBuilder) createBridge(
 		return 0, err
 	}
 
-	// Remove IP address from ENI link.
+	// If anything fails during setup, clean up the bridge so that the next call starts clean.
+	defer func() {
+		if err != nil {
+			log.Infof("Deleting bridge link %+v.", bridgeLink)
+			cleanupErr := netlink.LinkDel(bridgeLink)
```

Comment:
The [amazon-vpc-cni-k8s always deletes existing bridges on setup](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/plugins/routed-eni/driver/driver.go#L178), but this seems more efficient. 

Here we just return `bridge.Index` if the bridge exists, is that always safe? Will the `sharedENI` and `ipAddress` always be the same for the same bridge name?

Reply:
Yes. The bridge name includes the shared ENI's interface index, and is therefore unique on the instance. IP address is the shared ENI's primary IP address, and hence also unique.

Reply:
Ah, thanks for the explanation.